addons/mail/static/src/models/thread_view.js

```diff
@@ -267,6 +267,21 @@ registerModel({
             this.messaging.browser.clearTimeout(this.loaderTimeout);
             this.update({ isLoading: false, isPreparingLoading: false });
         },
+        /**
+         * @private
+         */
+        _onThreadChange() {
+            if (
+                this.messaging.discuss &&
+                this.messaging.discuss.thread === this.thread &&
+                this.messaging.discuss.discussView
+            ) {
+                this.env.services.router.pushState({
+                    action: this.messaging.discuss.discussView.actionId,
+                    active_id: this.messaging.discuss.activeId,
+                });
+            }
+        },
```

Comment:
Seeing these conditions, I think it would make sense to move this onChange to discuss view model. (so compared to master keep this part in discuss view, but move from on component update to onChange).